viewing_party/party.py

```diff
@@ -1,23 +1,286 @@
+        for friend_movie in friend["watched"]:
+            friend_movies_list.add(friend_movie["title"])
+    only_friends_watched_set = friend_movies_list.difference(user_movies_list)
+
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            if friend_movie["title"] in only_friends_watched_set and friend_movie not in only_friends_watched:
+                only_friends_watched.append(friend_movie)
+
+    return only_friends_watched
+
+
+def get_available_recs(user_data):
+    """Create a list of recommended movies for a user based on their subscriptions and friends' watched movies.
+
+    Parameters:
+        user_data (dict): 
+            A dictionary containing:
+            - "watched": list of movies the user has already watched
+            - "friends": list of friends, each with their own "watched" list
+            - "subscriptions": list of streaming services the user is subscribed to
+
+    Returns:
+        list: Movies that the user hasn't watched, that at least one friend has watched,
+            and are hosted on a service in the user's subscriptions.
+    """
+    
+    recommended_movies = []
+    
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+                if movie not in recommended_movies:
+                    recommended_movies.append(movie)
+                        
+    return recommended_movies
+
+
+def get_new_rec_by_genre(user_data):
+    """Determine a list of recommended movies
+
+    Parameters:
+        user_data (dict): 
+            A dictionary containing:
+            - "watched": list of movies the user has already watched
+            - "friends": list of friends, each with their own "watched" list
+
+    Returns:
+        list: list of recommended movies
+    """
+
+    most_watched_genre = get_most_watched_genre(user_data)
+    recommended_movies = []
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            if friend_movie not in user_data["watched"] and friend_movie["genre"] == most_watched_genre:
+                if friend_movie not in recommended_movies:
+                    recommended_movies.append(friend_movie)
+    return recommended_movies
+
+
+def get_rec_from_favorites(user_data):
+    """Determine a list of the user's favorite movies that none of the friends have watched
+    Parameters:
+        user_data (dict): 
+            A dictionary containing:
+            - "watched": list of movies the user has already watched
+            - "friends": list of friends, each with their own "watched" list
+            - "favorites": list of movie dictionaries
+
+    Returns:
+        list: list of recommended movies
+    """
+
+    rec_from_favorites = []
+    friend_titles_set= set()
+
+    if not user_data["friends"]:
+        return user_data["favorites"]

```

Comment:
The requirements don't state that if a user doesn't have any friends, then we should return the user's favorite movies. When following project requirements, if you have any questions or there are ambiguities related to what you need to do, please reach out to get clarity. 

❗️This will be especially important in industry since software developers do not determine business logic. Business logic is determined by product managers working often with finance teams to come up with a projects behaviors. 

Here, I would remove this check completely and let the function return an empty list of `rec_from_favorites` if a user has no friends.